agent/cmd/start.go

```diff
@@ -220,6 +222,7 @@ func newStartCommand(ctx context.Context, args []string, logger *logrus.Entry) *
 	viper.SetDefault(flagCacheDir, path.SystemCacheDir("sensu-agent"))
 	viper.SetDefault(flagDeregister, false)
 	viper.SetDefault(flagDeregistrationHandler, "")
+	viper.SetDefault(flagDetectCloudProvider, false)
```

Comment:
Have you considered leaving this enabled by default? Or is it too expensive to perform the detection functions?

Reply:
Sean and I kicked it back and forth for a bit. Here's the problem:

* There is no way to conclusively do cloud platform detection without sending at least one HTTP request to a metadata service.
* These requests typically go to a link-local IP address.
* Depending on the network, these requests could never complete once started.
* A timeout works, but makes agent startup take longer on non-cloud systems.
* Some users may not appreciate their agents making these network requests.

For this reason, we decided to put cloud provider detection behind a feature flag.

Reply:
Ah that makes sense, thanks for explaining!